sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/OkHttpAsyncHttpClient.java

```diff
@@ -178,15 +178,13 @@ private static RequestBody toOkHttpRequestBodySynchronously(BinaryData bodyConte
         }
 
         BinaryDataContent content = BinaryDataHelper.getContent(bodyContent);
-        if (content instanceof ByteArrayContent) {
+        if (content instanceof ByteArrayContent || content instanceof StringContent) {
```

Comment:
I'm guessing the old handler for `StringContent` did the same operations as `StringContent.toBytes()`?

Reply:
yeah, that's good point. I approximated what I found for netty here... I'll revert this.